### PR TITLE
Filling of gen TH3D + selection on zvtx (hard coded just for the time being)

### PR DESCRIPTION
**Pull request by arossi81**

### CODE REVIEW

**Comment by arossi81** [comment]
Hi @njacazio ,
we probably can directly filter collisions for zvtx (and other selections), but for the time being I added an hard-coded cut.

Ciao
Andrea

File: DPG/Tasks/qaEventTrackLite.cxx [hunk 1]
```diff
@@ -163,6 +163,14 @@ struct qaEventTrackLite {
   template <bool isMC, typename trackType>
   void fillHistograms(trackType const& track)
   {
+
+    if (TMath::Abs(track.dpgCollision().posZ() > 10.))
```
Comment (njacazio) [inline DPG/Tasks/qaEventTrackLite.cxx:167]: Could this be a configurable cut?

File: DPG/Tasks/qaEventTrackLite.cxx [hunk 2]
```diff
@@ -163,6 +163,14 @@ struct qaEventTrackLite {
...
+      return;
+    if constexpr (isMC) {
+      if (track.productionMode() == 0) {
+        histos.fill(HIST("Particle/selPtEtaPhiMCGenPrimary"), track.ptMC(), track.etaMC(), track.phiMC());
```
Comment (njacazio) [inline DPG/Tasks/qaEventTrackLite.cxx:171]: Would this not be filled twice? (line 286)

File: DPG/Tasks/qaEventTrackLite.cxx [hunk 3]
```diff
@@ -267,6 +275,9 @@ struct qaEventTrackLite {
     }
 
     for (const auto& particle : particles) {
+      if (TMath::Abs(particle.dpgCollision().posZ() > 10.))
```
Comment (njacazio) [inline DPG/Tasks/qaEventTrackLite.cxx:278]: Maybe we could use here the same configurable

**Review by njacazio — COMMENTED** [review]
Thanks Andrea! I have just minor suggestions

File: DPG/Tasks/qaEventTrackLite.cxx (hunk 3 above)
Comment (arossi81) [reply DPG/Tasks/qaEventTrackLite.cxx:278]: yes, of course, and also done via a Filter, but if you do not mind, I'll change it on Monday (or asap)

Comment (njacazio) [reply DPG/Tasks/qaEventTrackLite.cxx:278]: Ok!

Comment (njacazio) [reply DPG/Tasks/qaEventTrackLite.cxx]: I realized this was also wrong, the abs was on top of the bool selection, I fixed that in the last commit

Comment (arossi81) [reply DPG/Tasks/qaEventTrackLite.cxx]: ah, right thanks! Sorry. Instead, for the filling of the histo, that's correct. It's not the same histo and must be filled both in the loop over tracks and in that over nonreco particles.

File: DPG/Tasks/qaEventTrackLite.cxx (hunk 2 above)
Comment (arossi81) [reply DPG/Tasks/qaEventTrackLite.cxx:174]: This is the histo counting gen particles, it has to be filled both when looping over recoParticles and nonRecoparticles. Isn't that correct?

Comment (njacazio) [reply DPG/Tasks/qaEventTrackLite.cxx:174]: Ah yes, you're right, apologies!
Ok approved for me!

**Review by njacazio — APPROVED** [review]

**Comment by njacazio** [comment]
@mario-krueger I don't think I can approve yet, some git magic has to happen yet. Could you also have a look pls?

**Pull request merged**